src/local-mode.ts

```diff
@@ -142,6 +147,39 @@ export class LocalModeManager {
 		);
 	}
 
+	private async collectContainerResources(
+		nameOrId: string,
+	): Promise<EngineSnapshot> {
+		const inspectInfo = await this.docker.getContainer(nameOrId).inspect();
+		return new EngineSnapshot(
+			[inspectInfo.Id],
+			[inspectInfo.Image],
+			inspectInfo.Mounts.filter(m => m.Name !== undefined).map(m => m.Name!),
+			_.values(inspectInfo.NetworkSettings.Networks).map(ns => ns.NetworkID),
+		);
+	}
+
+	// Determine what engine objects are linked to our own container.
+	private async collectOwnResources(): Promise<EngineSnapshot> {
+		try {
+			return this.collectContainerResources(
+				this.containerId || SUPERVISOR_CONTAINER_NAME_FALLBACK,
+			);
+		} catch (e) {
+			if (this.containerId !== undefined) {
+				// Inspect operation fails (container ID is out of sync?).
+				const fallback = SUPERVISOR_CONTAINER_NAME_FALLBACK;
+				log.error(
+					'Supervisor container resources cannot be obtained by container ID. ' +
+						`Using '${fallback}' name instead.`,
+					e.message,
+				);
+				return this.collectContainerResources(fallback);
+			}
+			throw e;
```

Comment:
Can you make this a typed error please? Often errors can come from seemingly nowhere, so wrapping this would help know it's origin.

Reply:
The problem here is that I don't think dockerode/docker-modem throws a typed error.
https://github.com/apocas/docker-modem/blob/master/lib/modem.js#L257

Reply:
Or do you mean wrapping it with our own type?

Reply:
Wrapping it with our type, for example
```typescript
import * as TypedError from 'typed-error';
export class SupervisorContainerNotFoundError extends TypedError {}

throw new SupervisorContainerNotFoundError(e)
```

Reply:
normal typescript import does not work (lint fails), so I used an approach from other files (like `src/lib/errors.ts`):
```
import TypedError = require('typed-error');
```
I believe the package needs an upgrade, but this should be done in a separate PR.

Reply:
Ah yeah I forgot about that - I was waiting for 100% typescript before doing that.